### PR TITLE
Add configuration APIs for Certificate configuration

### PR DESCRIPTION
**Pull request by prkalle**
### What this PR does / why we need it
This PR is to add configuration APIs for custom certificate configuration

The sample configuration is shown below which would be added to the configuration file (config-ng.yaml)

```
certs:
    - hostName: localhost:9876
      skipCertVerify: "true"
      insecure: "true"
      caCertData: <REDACTED>
```
### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

Unit tests and CI passed

### Release note

```release-note
Add configuration APIs for Certificate configuration
```

### CODE REVIEW

File: config/certs.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+		return nil, err
+	}
+	if cfg.Certs != nil {
+		return cfg.Certs, nil
+	}
+	return nil, errors.New("certs not found")
+}
+
+func getCert(node *yaml.Node, hostName string) (*configtypes.Cert, error) {
+	cfg, err := convertNodeToClientConfig(node)
```
Comment (mpanchajanya) [inline config/certs.go:99]: nit; pls add hostName validation steps in Get/Delete/Set methods especially if its value is "". so we don't end up have invalid certs like certs without hostname

File: config/certs_test.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+func TestSetCert(t *testing.T) {
+	// Setup config data
+	_, cleanUp := setupTestConfig(t, &CfgTestData{})
+
+	defer func() {
+		cleanUp()
+	}()
+
+	tests := []struct {
```
Comment (mpanchajanya) [inline config/certs_test.go:72]: pls add few negative test cases where Set/Get/Delete return errors
when invalid cert data is used to set/delete/get

**Review by mpanchajanya — CHANGES_REQUESTED** [review]
Changes LGTM!. nit! on validating the input args and negative test cases

File: config/certs.go (hunk 1 above)
Comment (prkalle) [reply config/certs.go:99]: Good point. Updated. Thanks

File: config/certs_test.go (hunk 2 above)
Comment (prkalle) [reply config/certs_test.go:72]: Updated.

**Review by mpanchajanya — APPROVED** [review]

**Pull request merged**